src/service.rs

```diff
@@ -18,21 +20,21 @@ where
     S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
     B: MessageBody,
 {
-    type Response = S::Response;
+    type Response = ServiceResponse<EitherBody<B>>;
     type Error = S::Error;
-    type Future =
-        future::Either<future::Ready<Result<ServiceResponse<B>, actix_web::Error>>, S::Future>;
+    type Future = Either<
+        MapOk<S::Future, fn(ServiceResponse<B>) -> ServiceResponse<EitherBody<B>>>,
+        Ready<Result<ServiceResponse<EitherBody<B>>, Self::Error>>,
+    >;
```

Comment:
Can you make this a bit more readable? I wouldn't be surprised if clippy complained about type complexity here.

Reply:
Added type alias for MapOk part.